packages/commerce-sdk-react/src/hooks/ShopperBaskets/helpers.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useCustomerId, useShopperBasketsMutation} from '../index'
+import {useCustomerBaskets} from '../ShopperCustomers'
+import {ApiClients, Argument} from '../types'
+import {ShopperBasketsTypes} from 'commerce-sdk-isomorphic'
+type Client = ApiClients['shopperBaskets']
+type Basket = ShopperBasketsTypes.Basket
+
+/**
+ * This is a helper function for Basket Mutations.
+ * useShopperBasketsMutationHelper.addItemToNewOrExistingBasket: is responsible for managing the process of adding an item to a basket.
+ *  - If a basket already exists, add the item to the basket immediately.
+ *  - If a basket does not exist, create a new basket using the createBasket mutation
+ * and then add the item to the newly created basket using the addItemToBasket mutation.
+ *
+ * @example
+ * import useShopperBasketsMutationHelper from '@salesforce/commerce-sdk-react'
+ *
+ * const Page = () => {
+ *      const helpers = useShopperBasketsMutationHelper()
+ *
+ *      const addToCart = async () => {
+ *          const productItems = [{id: 123, quantity: 2}]
+ *          await basketMutationHelpers.addItemToNewOrExistingBasket(productItems)
+ *      }
+ *
+ * }
+ */
+export function useShopperBasketsMutationHelper() {
+    const customerId = useCustomerId()
+    const {data: basketsData} = useCustomerBaskets(
+        {parameters: {customerId}},
+        {
+            enabled: !!customerId
+        }
+    )
+    const createBasket = useShopperBasketsMutation('createBasket')
+    const addItemToBasketMutation = useShopperBasketsMutation('addItemToBasket')
+    return {
+        addItemToNewOrExistingBasket: async (
+            productItem: Argument<Client['addItemToBasket']> extends {body: infer B} ? B : undefined
+        ): Promise<Basket> => {
+            if (basketsData && basketsData.total > 0) {
+                // we know that if basketData.total > 0, current basket will always be available
```

Comment:
Is it possible to still have an empty basket that exists?

ie. I add an item to my basket, which creates a new basket if I don't have one. Then I delete the item so now I have an empty basket. What happens when I try to add another item?

Reply:
I misunderstood. This code is checking if basket data exists not if baskets are holding onto lineitems.